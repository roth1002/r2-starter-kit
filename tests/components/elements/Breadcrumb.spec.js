import React from 'react';
import { shallow } from 'enzyme';
import Component from 'components/elements/Breadcrumb';

describe('<Breadcrumb />', () => {
  const initProps = {
    data: [],
    activeId: null,
  };

  const props = {
    data: [
      { id: -1, text: 'Home', handleClick: jest.fn() },
      { id: 1, text: 'title', handleClick: () => {} },
    ],
    activeId: 1,
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Component {...initProps} />);
  });

  it('should render initial component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with props', () => {
    wrapper.setProps(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleClick when click root path', () => {
    expect(props.data[0].handleClick).toHaveBeenCalledTimes(0);
    wrapper.find('a').simulate('click');
    expect(props.data[0].handleClick).toHaveBeenCalledTimes(1);
  });
});
