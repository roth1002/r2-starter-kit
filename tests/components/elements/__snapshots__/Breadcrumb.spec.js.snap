// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render component with props 1`] = `
<div
  className="ts breadcrumb"
>
  <div
    className="section"
    key="breadcrumb0"
  >
    <a
      onClick={[MockFunction]}
    >
      Home
    </a>
  </div>
  <div
    className="section active"
    key="breadcrumb1"
  >
    <i
      className="right angle icon divider"
    />
    title
  </div>
</div>
`;

exports[`<Breadcrumb /> should render initial component 1`] = `
<div
  className="ts breadcrumb"
/>
`;
